### PR TITLE
[FLINK-36038][table] Output for `SHOW CREATE VIEW` should follow the `CREATE VIEW` syntax

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

1. `COMMENT` in `SHOW CREATE VIEW` should be before the query to satisfy `CREATE VIEW` syntax [1]
2. Make output of `SHOW CREATE VIEW` be similar to `SHOW CREATE TABLE`
2.1. every column on its own line
2.2 there should be a space between view name and parentheses
2.3 `as` keyword should be uppercase `AS`
2.4 it should finish with new line (like for `SHOW CREATE TABLE`) 

[1] https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/create/#create-view

## Verifying this change

There is new `ShowCreateUtilTest.java` and a number of existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/internal/ShowCreateUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+    private static final ResolvedSchema TWO_COLUMNS_SCHEMA =
+            ResolvedSchema.of(
+                    Column.physical("id", DataTypes.INT()),
+                    Column.physical("name", DataTypes.STRING()));
+
+    @ParameterizedTest(name = "{index}: {1}")
+    @MethodSource("argsForShowCreateTable")
+    void showCreateTable(ResolvedCatalogTable resolvedCatalogTable, String expected) {
+        final String createViewString =
```
Comment (twalthr) [inline flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/internal/ShowCreateUtilTest.java:63]:
```suggestion
        final String createTableString =
```

File: flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/internal/ShowCreateUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                        createResolvedTable(
+                                ONE_COLUMN_SCHEMA,
+                                Collections.emptyMap(),
+                                Collections.emptyList(),
+                                TableDistribution.of(
+                                        TableDistribution.Kind.RANGE, 2, Arrays.asList("1", "10")),
+                                "Table comment"),
+                        "CREATE TABLE `catalogName`.`dbName`.`tableName` (\n"
+                                + "  `id` INT\n"
+                                + ") COMMENT 'Table comment'\n"
```
Comment (twalthr) [inline flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/internal/ShowCreateUtilTest.java:129]: It's weird that COMMENT causes a newline at the end whereas other subclauses don't. We should fix this or we will never fix it. Can we add a `\n` after `)`:

```
CREATE TABLE `catalog`.`database`.`t` (
  `s` VARCHAR(2147483647)
)
DISTRIBUTED INTO 6 BUCKETS
WITH (
  'k' = 'v'
)
```

```
CREATE TABLE `catalog`.`database`.`t` (
  `s` VARCHAR(2147483647)
)
COMMENT 'comment'
DISTRIBUTED INTO 6 BUCKETS
WITH (
  'k' = 'v'
)
```

```
CREATE TABLE `catalog`.`database`.`t` (
  `s` VARCHAR(2147483647)
)
COMMENT 'comment'
DISTRIBUTED INTO 6 BUCKETS
PARTITIONED BY (`key1`, `key2`)
WITH (
  'k' = 'v'
)
```

```
CREATE VIEW `catalogName`.`dbName`.`viewName` (
  `id`
  `name`
)
COMMENT 'View comment'
AS SELECT ...
```

**Comment by snuyanzin** [comment]
@flinkbot run azure

**Comment by snuyanzin** [comment]
@flinkbot run azure

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java [hunk 3]
```diff
@@ -176,10 +161,7 @@ static String getColumnString(Column column) {
                         comment -> {
                             if (StringUtils.isNotEmpty(comment)) {
                                 sb.append(" ");
```
[the remark this answers is not in the document]
Comment (twalthr) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java:163]: nvm, I thought this is the table comment

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java [hunk 4]
```diff
@@ -246,13 +230,19 @@ static Optional<String> extractFormattedOptions(Map<String, String> conf, String
                                                 printIndent,
                                                 EncodingUtils.escapeSingleQuotes(entry.getKey()),
                                                 EncodingUtils.escapeSingleQuotes(entry.getValue())))
+                        .sorted()
```
[the remark this answers is not in the document]
Comment (twalthr) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java:233]: Found the answer already.

**Review by twalthr — APPROVED** [review]
Thanks for making Flink more consistent.

**Pull request merged**